packages/xdg_directories/lib/xdg_directories.dart

```diff
@@ -49,12 +49,12 @@ set xdgProcessManager(ProcessManager processManager) {
 ProcessManager _processManager = const LocalProcessManager();
 
 List<Directory> _directoryListFromEnvironment(
-    String envVar, List<Directory> fallback) {
+    String? envVar, List<Directory>? fallback) {
```

Comment:
Same here for `envVar`. Nor do I see an argument for allowing null fallback.

And in fact the next two lines assert that they aren't null, so this makes even less sense.

Reply:
If I don't make them nullable then dart shows a unnecessary null comparison warning on the assert condition. (`The operand can't be null, so the condition is always true.
Remove the condition`)

Reply:
https://dart.dev/null-safety/faq#how-should-i-migrate-an-assertx--null-that-now-shows-as-unnecessary

Reply:
I have made these changes.